src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -84,6 +96,38 @@ public String updateUserEmail(UserChangeEmailDto userChangeEmailDto, User user)
         return getMessageSource("changed.user.email");
     }
 
+    public String deleteUserFirst(User user) {
+        user.setStatus(DELETED);
+        userRepository.saveAndFlush(user);
+
+        return getParametrizedMessageSource("account.deleted.first", getDaysBeforeDeletion(user));
```

Comment:
You're ruined SOLID's principle "Single responsibility" here - you definitely need to update it.

Reply:
https://user-images.githubusercontent.com/49348448/109987285-2a254880-7d0f-11eb-9b69-98ab5cb0dfb6.png
Done. But we have the same solution in 2 methods which has been approved recently. Should I fix them? If yes, then within the existing task?